autoPyTorch/api/base_task.py

```diff
@@ -190,6 +194,8 @@ def __init__(
         self.pipeline_options = replace_string_bool_to_bool(json.load(open(
             os.path.join(os.path.dirname(__file__), '../configs/default_pipeline_options.json'))))
 
+        self._visualizer = ResultsVisualizer()
```

Comment:
Why do we need it to be an attribute of self? Do you expect to reuse an already created instance of this class?  Unlike `results_manager`, this class does not hold any data. Also, I don't expect a user to call `plot_perf_over_time` more than once with the same parameters. So I suggest you recreate an instance whenever the `plot_perf_over_time` function is called.

Reply:
+1